doc/rtd/howto/return_codes.rst

```diff
@@ -0,0 +1,172 @@
+.. _return_codes:
+
+Why did `cloud-init status` start returning exit code 2?
+========================================================
+
+Cloud-init introduced a new error code in 23.4.
+This page describes the purpose of this change and gives
+some context for why this change was made.
+
+.. _return_codes_history:
+
+History
+-------
+
+In the early days of cloud-init, the purpose of cloud-init
+was simple: provision ssh keys and a user and exit. Cloud-init
+only supported a couple of clouds, and things were simple.
+Since cloud-init provides access to cloud instances, the
```

Comment:
I don't think we need the sentences before this. I don't care about the early days of cloud-init, more just the reasoning behind why things work the way they do.